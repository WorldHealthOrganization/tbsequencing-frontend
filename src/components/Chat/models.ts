export interface IUserMessage {
  email: string,
  username: string,
  firstName: string,
  lastName: string
}
