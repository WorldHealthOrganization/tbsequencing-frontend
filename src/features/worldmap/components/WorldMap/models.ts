export type MapType = 'macroRegions' | 'heatmap';
