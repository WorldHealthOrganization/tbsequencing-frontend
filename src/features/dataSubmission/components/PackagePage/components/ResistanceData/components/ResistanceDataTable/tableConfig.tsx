/** @jsxImportSource @emotion/react */
import React from 'react';
import { match } from 'ts-pattern';
import { createColumnHelper } from '@tanstack/react-table';
import { IDataSample, IVerdict, MatchSource } from '../../../../../../../../services/submissionApi/models';
import PrimaryText from '../../../../../../../../components/typography/PrimaryText';
import { Nullable } from '../../../../../../../../models/tsHelpers';
import EditableSampleName from '../../../EditableSampleName';
import ResistanceFileVerdict from '../../../ResistanceFileVerdict';

type TableHeaderCellProps = {
  label: string;
};

const columnHelper = createColumnHelper<IDataSample>();

const TableHeaderCell = ({ label }: TableHeaderCellProps) => (
  <PrimaryText>{label}</PrimaryText>
);

export const getTableColumns = (packageId: number, controlsDisabled: boolean) => [
  columnHelper.accessor('name', {
    cell: ({ row, getValue }) => {
      const name = getValue();
      const sampleId = row.original.pk;

      return (
        <EditableSampleName
          controlsDisabled={controlsDisabled}
          packageId={packageId}
          sampleId={sampleId}
          name={name}
        />
      );
    },
    header: () => <TableHeaderCell label="Sample Name" />,
  }),
  columnHelper.accessor('verdicts', {
    cell: ({ getValue }) => {
      const verdicts = getValue();

      if (!verdicts || verdicts.length === 0) {
        return null;
      }

      const verdict = verdicts[0] as IVerdict;
      // as we have always one verdict we have to take single verdict

      if (!verdict) {
        return null;
      }

      return (
        <ResistanceFileVerdict verdict={verdict} />
      );
    },
    header: () => <TableHeaderCell label="Verdicts" />,
  }),
  columnHelper.accessor('fastqPrefix', {
    cell: ({ getValue }) => {
      const prefix = getValue() as string;

      return <PrimaryText>{prefix}</PrimaryText>;
    },
    header: () => <TableHeaderCell label="FastQ prefix" />,
  }),
  columnHelper.accessor('matchSource', {
    cell: ({ getValue }) => {
      const matchSource = getValue();
      const cellLabel = match<Nullable<MatchSource>, string>(matchSource)
        .with(null, () => '')
        .with('NO_MATCH', () => 'Match not found')
        .with('NCBI', () => 'Matched with NCBI/EBI data')
        .with('FASTQ_EXISTING', () => 'Matched with existing FASTQ')
        .with('FASTQ_UPLOADED', () => 'Matched with uploaded FASTQ')
        .with('FASTQ_UPLOADED_NEW_SAMPLE', () => 'New sample created from uploaded FASTQ')
        .with('USER_ALIAS', () => 'Matched with sample previously uploaded by yourself')
        .with('NCBI_FASTQ', () => 'Existing FASTQ file at NCBI/EBI')
        .exhaustive();

      return <PrimaryText>{cellLabel}</PrimaryText>;
    },
    header: () => <TableHeaderCell label="Match source" />,
  }),
];
