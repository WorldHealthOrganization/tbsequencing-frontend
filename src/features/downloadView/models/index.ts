import { IDrug } from '../../../services/drugsApi/models';

export interface IExtendedDrug extends IDrug {
  isChecked: boolean;
}
