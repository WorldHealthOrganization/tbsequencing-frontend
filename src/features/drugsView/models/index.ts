export enum DrugStatus {
  Susceptible = 'susceptible',
  Resistant = 'resistant',
  Intermediate = 'intermediate',
  RatioSusceptible = 'ratioSusceptible',
  RatioResistant = 'ratioResistant',
  RatioIntermediate = 'ratioIntermediate',
}

export enum ResistantType {
  Phenotypic = 'Pheno',
  Genotypic = 'Geno',
}

export enum ChartDataType {
  Ratio = 'ratio',
  Count = 'count',
}

export enum Regions {
  ALL = 'ALL_REGIONS',
  AFRICA = 'AFR',
  EUROPE = 'EUR',
  AMERICA = 'AMR',
  EASTERN_MEDITERRANEAN = 'EMR',
  WESTERN_PACIFIC = 'WPR',
  SOUTH_EAST_ASIA = 'SEA',
}
