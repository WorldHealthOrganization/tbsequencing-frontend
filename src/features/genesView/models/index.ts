import { skipToken } from '@reduxjs/toolkit/query';
import React from 'react';
import { IGene, IVariant } from '../../../services/genesApi/models';

export type AutocompleteParams = typeof skipToken | { search?: string };

export interface IGenomeContextWrapperProps {
  data?: IGene[];
  isLoading?: boolean;
  selectedGene?: IGenesOverview;
  minDomain?: number;
  maxDomain?: number;
}

export interface IGenomeContextProps extends IGenomeContextWrapperProps {
  width: number;
}

export interface IVariantLocal extends IVariant {
  pos: number;
}

export interface ISearchResult {
  geneName: string;
  geneDbCrossrefId: number;
  locusTag: string;
}

export interface ISearchCount extends ISearchResult {
  counter: number;
}

export interface ISearchRecent extends ISearchResult {
  counter: string;
  date: Date | string;
}

export interface IGenesOverview extends IGene {
  geneDescription: string;
  geneType: string;
  ncbiId: string;
  proteinLength: number;
}

export interface IGenesOverviewTableData {
  title: string;
  value: string | number | null;
  element?: React.ReactNode;
}

export interface IGenesOverviewResponse {
  genes: IGene;
  genesOverview: IGenesOverview
}
