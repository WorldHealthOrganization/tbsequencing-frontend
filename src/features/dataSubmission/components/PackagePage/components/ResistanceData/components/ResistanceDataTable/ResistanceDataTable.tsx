/** @jsxImportSource @emotion/react */
import React, { useMemo, useState } from 'react';
import {
  ExpandedState, flexRender, getCoreRowModel, getExpandedRowModel, useReactTable,
} from '@tanstack/react-table';
import {
  Table, TableBody, TableCell, TableContainer, TableHead, TableRow,
} from '@mui/material';
import { IDataSample, ResistanceSampleType } from '../../../../../../../../services/submissionApi/models';
import PackageDataGridCell from '../../../PackageDataGridCell';
import ResistanceDataSamplesHeaderRow from '../ResistanceDataHeaderRow';
import { getTableColumns } from './tableConfig';
import * as styles from '../../../../../SubmissionPackageList/components/SubmissionPackageListTable/styles';

interface Props {
  samplesData: IDataSample[];
  title: string;
  packageId: number;
  controlsDisabled: boolean;
  activeTab: ResistanceSampleType;
}

export const ResistanceDataTable = ({
  samplesData, title, packageId, controlsDisabled, activeTab,
}: Props) => {
  const [expanded, setExpanded] = useState<ExpandedState>(true);
  const tableColumns = useMemo(() => getTableColumns(
    packageId,
    controlsDisabled,
  ), [packageId, controlsDisabled, activeTab]);

  const instance = useReactTable({
    data: samplesData,
    columns: tableColumns,
    getCoreRowModel: getCoreRowModel(),
    onExpandedChange: setExpanded,
    getExpandedRowModel: getExpandedRowModel(),
    state: {
      expanded,
    },
  });

  if (!samplesData) {
    return null;
  }

  const handleExpandClick = () => {
    instance.toggleAllRowsExpanded();
  };

  return (
    <>
      <ResistanceDataSamplesHeaderRow
        onExpandClick={handleExpandClick}
        title={title}
        totalItems={samplesData.length}
      />
      <TableContainer>
        <Table>
          <TableHead>
            <TableRow>
              {instance.getFlatHeaders().map((header) => {
                if (expanded !== true) {
                  return null;
                }

                return (
                  <TableCell key={header.column.columnDef.id} css={styles.tableHeaderCell}>
                    {
                      flexRender(
                        header.column.columnDef.header,
                        header.getContext(),
                      )
                    }
                  </TableCell>
                );
              })}
            </TableRow>
          </TableHead>
          <TableBody>
            {instance.getRowModel().rows.map((row) => {
              const isExpanded = row.getIsExpanded();

              if (!isExpanded) {
                return null;
              }

              return (
                <TableRow>
                  {
                    row.getVisibleCells().map((cell) => (
                      <PackageDataGridCell>
                        {flexRender(cell.column.columnDef.cell, cell.getContext())}
                      </PackageDataGridCell>
                    ))
                  }
                </TableRow>
              );
            })}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  );
};

export default ResistanceDataTable;
