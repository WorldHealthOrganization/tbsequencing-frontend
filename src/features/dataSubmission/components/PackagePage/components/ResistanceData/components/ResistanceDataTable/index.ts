import ResistanceDataTable from './ResistanceDataTable';

export default ResistanceDataTable;
