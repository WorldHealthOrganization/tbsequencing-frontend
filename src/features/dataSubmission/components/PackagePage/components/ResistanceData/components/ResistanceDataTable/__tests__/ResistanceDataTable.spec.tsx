import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../../../../../../../utils/testUtils/jestUtils';
import ResistanceDataTable from '../ResistanceDataTable';
import { IDataSample } from '../../../../../../../../../services/submissionApi/models';

describe('ResistanceDataTable component test', () => {
  const mockSamples: IDataSample[] = [{
    name: 'mock',
    pk: 0,
    url: 'mock',
    micTestsCount: 0,
    pdsTestsCount: 0,
    createdAt: '2022-09-07T21:01:49.455163Z',
    matchSource: 'FASTQ_UPLOADED_NEW_SAMPLE',
    fastqPrefix: 'MOCK',
    verdicts: [{ level: 'warning', verdict: 'mock' }],
    sample: {
      additionalGeographicalInformation: '',
      biosampleId: 0,
      country: 'mock',
      isolationSource: 'mock',
      latitude: 0,
      longitude: 0,
      patiendId: 0,
      samplingDate: '2022-09-07',
      submissionDate: '2022-09-07',
    },
  }];

  test('renders ResistanceDataTable component without error', () => {
    render(
      <ResistanceDataTable samplesData={mockSamples} title="mockTitle" />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <ResistanceDataTable samplesData={mockSamples} title="mockTitle" />,
    );

    expect(tree).toMatchSnapshot();
  });
});
