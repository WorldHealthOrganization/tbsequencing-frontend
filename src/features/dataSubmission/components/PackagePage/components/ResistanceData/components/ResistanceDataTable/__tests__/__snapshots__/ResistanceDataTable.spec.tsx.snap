// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResistanceDataTable component test Should render correctly 1`] = `
Array [
  <div
    className="css-1udyuk7-ResistanceDataSamplesHeaderRow"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1134uzf-MuiTypography-root-BaseText"
    >
      mockTitle
    </p>
    <div
      className="css-zuupt0-ResistanceDataSamplesHeaderRow"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-9wau7z-MuiTypography-root-BaseText"
      >
        1 items
      </p>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="expand-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1mqn5fw-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </button>
    </div>
  </div>,
  <div
    className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root css-rqglhn-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nghee1-MuiTableCell-root-ResistanceDataTable"
            scope="col"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              Sample Name
            </p>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nghee1-MuiTableCell-root-ResistanceDataTable"
            scope="col"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              Verdicts
            </p>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nghee1-MuiTableCell-root-ResistanceDataTable"
            scope="col"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              FastQ prefix
            </p>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nghee1-MuiTableCell-root-ResistanceDataTable"
            scope="col"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              Match source
            </p>
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-qo7vjo-MuiTableCell-root-PackageDataGridCell"
          >
            <div
              className="css-1actu8n-EditableSampleName"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
              >
                mock
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1s1qiw1-MuiSvgIcon-root"
                  data-testid="ModeEditOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 21h3.75L17.81 9.94l-3.75-3.75L3 17.25V21zm2-2.92 9.06-9.06.92.92L5.92 19H5v-.92zM18.37 3.29a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34z"
                  />
                </svg>
              </button>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-qo7vjo-MuiTableCell-root-PackageDataGridCell"
          >
            <div
              className="css-icwgla-ResistanceFileVerdict"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-v1629q-MuiSvgIcon-root"
                data-testid="WarningAmberOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 5.99 19.53 19H4.47L12 5.99M12 2 1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                />
              </svg>
              <p
                className="MuiTypography-root MuiTypography-body1 css-phgkfg-MuiTypography-root-BaseText"
              >
                mock
              </p>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-qo7vjo-MuiTableCell-root-PackageDataGridCell"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              MOCK
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-qo7vjo-MuiTableCell-root-PackageDataGridCell"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
            >
              New sample created from uploaded FASTQ
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
